newrelic/hooks/framework_strawberry.py

```diff
@@ -0,0 +1,123 @@
+
+
+def framework_details():
+    import strawberry
+
+    return ("Strawberry", getattr(strawberry, "__version__", None))
+
+
+def bind_execute(query, *args, **kwargs):
+    return query
+
+
+def wrap_execute_sync(wrapped, instance, args, kwargs):
+    transaction = current_transaction()
+
+    if not transaction:
+        return wrapped(*args, **kwargs)
+
+    try:
+        query = bind_execute(*args, **kwargs)
+    except TypeError:
+        return wrapped(*args, **kwargs)
+
+    framework = framework_details()
+    transaction.add_framework_info(name=framework[0], version=framework[1])
+    transaction.add_framework_info(name="GraphQL", version=graphql_framework_version())
+
+    if hasattr(query, "body"):
+        query = query.body
+
+    transaction.set_transaction_name(callable_name(wrapped), "GraphQL", priority=10)
+
+    with GraphQLOperationTrace() as trace:
+        trace.statement = graphql_statement(query)
+        with ErrorTrace(ignore=ignore_graphql_duplicate_exception):
+            return wrapped(*args, **kwargs)
+
+
+async def wrap_execute(wrapped, instance, args, kwargs):
+    transaction = current_transaction()
+
+    if not transaction:
+        return await wrapped(*args, **kwargs)
+
+    try:
+        query = bind_execute(*args, **kwargs)
+    except TypeError:
+        return await wrapped(*args, **kwargs)
+
+    framework = framework_details()
+    transaction.add_framework_info(name=framework[0], version=framework[1])
+    transaction.add_framework_info(name="GraphQL", version=graphql_framework_version())
+
+    if hasattr(query, "body"):
+        query = query.body
+
+    transaction.set_transaction_name(callable_name(wrapped), "GraphQL", priority=10)
+
+    with GraphQLOperationTrace() as trace:
+        trace.statement = graphql_statement(query)
+        with ErrorTrace(ignore=ignore_graphql_duplicate_exception):
+            return await wrapped(*args, **kwargs)
+
+
+def bind_from_resolver(field, *args, **kwargs):
+    return field
+
+
+def wrap_from_resolver(wrapped, instance, args, kwargs):
+    result = wrapped(*args, **kwargs)
+    field = bind_from_resolver(*args, **kwargs)
+
+    if hasattr(field, "base_resolver"):
+        if hasattr(field.base_resolver, "wrapped_func"):
+            resolver_name = callable_name(field.base_resolver.wrapped_func)
+            result = TransactionNameWrapper(result, resolver_name, "GraphQL", priority=13)
+
+    return result
+
+
+def instrument_strawberry_schema(module):
+    if hasattr(module, "Schema"):
+        if hasattr(module.Schema, "execute"):
+            wrap_function_wrapper(module, "Schema.execute", wrap_execute)
+        if hasattr(module.Schema, "execute_sync"):
+            wrap_function_wrapper(module, "Schema.execute_sync", wrap_execute_sync)
+
+
+def instrument_strawberry_asgi(module):
+    if hasattr(module, "GraphQL"):
+        wrap_asgi_application(module, "GraphQL.__call__", framework=framework_details())
```

Comment:
Do we need to add the framework details here as well as on the execute transaction or just one place?